extras/gha_build.sh

```diff
@@ -0,0 +1,352 @@
+    # will cause the Travis CI build to fail or are ignored.
+    IGNORE_MSG_VERA=false
+    IGNORE_MSG_CPPCHECK=true
+    IGNORE_MSG_CLANG_FORMAT=false
+    IGNORE_MSG_PEP8=false
+
+    # The script is called within the Travis CI environment and thus can not be run incremental.
+    #RUNS_ON_TRAVIS=true
+    INCREMENTAL=false
+
+    chmod +x extras/static_code_analysis.sh
+    ./extras/static_code_analysis.sh "$RUNS_ON_TRAVIS" "$INCREMENTAL" "$file_names" "$NEST_VPATH" \
+    "$VERA" "$CPPCHECK" "$CLANG_FORMAT" "$PEP8" \
+    "$PERFORM_VERA" "$PERFORM_CPPCHECK" "$PERFORM_CLANG_FORMAT" "$PERFORM_PEP8" \
+    "$IGNORE_MSG_VERA" "$IGNORE_MSG_CPPCHECK" "$IGNORE_MSG_CLANG_FORMAT" "$IGNORE_MSG_PEP8"
+
+    exit $?
+fi
+
+echo
+echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
+echo "+               C O N F I G U R E   N E S T   B U I L D                       +"
+echo "+ + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + + +"
+echo "MSGBLD0230: Reading build type."
+
+# This defines the base settings of all build options off. The base
+# settings are also used for NEST_BUILD_TYPE=MINIMAL, which is not
+# explicitly checked for below.
+
+xGSL=0
+xLIBBOOST=0
+xLIBNEUROSIM=0
+xLTDL=0
+xMPI=0
+xMUSIC=0
+xOPENMP=0
+xPYTHON=0
+xREADLINE=0
+xSIONLIB=0
+
+CXX_FLAGS="-pedantic -Wextra -Wno-unknown-pragmas"
+
+if [ "$xNEST_BUILD_TYPE" = "OPENMP_ONLY" ]; then
+    xGSL=1
+    xLIBBOOST=1
+    xLTDL=1
+    xOPENMP=1
+    CXX_FLAGS="-pedantic -Wextra"
+fi
+
+if [ "$xNEST_BUILD_TYPE" = "MPI_ONLY" ]; then
+    xGSL=1
+    xLIBBOOST=1
+    xLTDL=1
+    xMPI=1
+fi
+
+if [ "$xNEST_BUILD_TYPE" = "FULL" ]; then
+    xGSL=1
+    xLIBBOOST=1
+    xLIBNEUROSIM=1
+    xLTDL=1
+    xMPI=1
+    xMUSIC=1
+    xOPENMP=1
+    xPYTHON=1
+    xREADLINE=1
+    xSIONLIB=1
+    CXX_FLAGS="-pedantic -Wextra"
+fi
+
+if [ "$xNEST_BUILD_TYPE" = "FULL_NO_EXTERNAL_FEATURES" ]; then
+    xGSL=1
+    xLIBBOOST=1
+    xLIBNEUROSIM=0
+    xLTDL=1
+    xMPI=0
+    xMUSIC=0
+    xOPENMP=1
+    xPYTHON=1
+    xREADLINE=1
+    xSIONLIB=0
+fi
+
+echo "MSGBLD0232: Setting configuration variables."
+
+# Set the NEST CMake-build configuration according to the variables
+# set above based on the ones set in the build stage matrix in
+# '.travis.yml'.
+
+if [ "$xOPENMP" = "1" ] ; then
+    CONFIGURE_OPENMP="-Dwith-openmp=ON"
+else
+    CONFIGURE_OPENMP="-Dwith-openmp=OFF"
+fi
+
+if [ "$xMPI" = "1" ] ; then
+    CONFIGURE_MPI="-Dwith-mpi=ON"
+else
+    CONFIGURE_MPI="-Dwith-mpi=OFF"
+fi
+
+if [ "$xPYTHON" = "1" ] ; then
+    export PYTHON_INCLUDE_DIR=`python3 -c "import sysconfig; print(sysconfig.get_path('include'))"`
+    export PYLIB_BASE=lib`basename $PYTHON_INCLUDE_DIR`
+    export PYLIB_DIR=$(dirname `sed 's/include/lib/' <<< $PYTHON_INCLUDE_DIR`)
+    export PYTHON_LIBRARY=`find $PYLIB_DIR \( -name $PYLIB_BASE.so -o -name $PYLIB_BASE.dylib \) -print -quit`
```

Comment:
I'm impressed to see that our detection mechanism is even stable enough to work on a completely different CI system!

:smile_cat:

Reply:
> Although I think that our internal variables (e.g. `xNEST_BUILD_TYPE`) would be fine without the leading `x` and others (e.g. `xGSL`) should better be called `NEST_BUILD_FLAG_...` I think we should merge this quickly and improve on such things later on.
> 
> In this sense: full approval and again many, many thanks for your hard work and the uncounted iterations this one took you.

Thank you very much! :-)